### ISSUE TITLE
context.setLocale is not working

### ISSUE THREAD

**Issue by mauryagaurav947** [aissat/easy_localization#552]
In my case context.setLocale is not working until i hotreload the application

**Comment by bw-flagship** [comment]
Could you provide a minimal sample to reproduce?

**Comment by mauryagaurav947** [comment]
Until I reload the app the language change does not affect it.

For e.g. have a look at the video link
https://drive.google.com/file/d/1trNPZ0IIcNV7ADu_QL60fv1K8cx5-no1/view?usp=share_link

**Comment by bw-flagship** [comment]
That's a video, not a sample to reproduce.
Remember to call setState after changing the language.

**Comment by mauryagaurav947** [comment]
Thanks a lot @bw-flagship for your support 😎

**Comment by aissat** [comment]
Just pass `context` in `tr()`  one time is enough

`Text("title").tr(context: context)`
// or 
`tr("title",context: context)`

**Comment by ckp-aayushi** [comment]
> Just pass `context` in `tr()` one time is enough
> 
> `Text("title").tr(context: context)` // or `tr("title",context: context)`

I am getting error on using this: 

The named parameter 'context' isn't defined.
Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'context'.dart[undefined_named_parameter](https://dart.dev/diagnostics/undefined_named_parameter)

Easy-localization version: ^3.0.1

Can you pls let me know in which version context is available?

**Comment by omensight** [comment]
I have passed the context to the tr() method but that didn't work. Help please

**Comment by bw-flagship** [comment]
@omensight please post code to reproduce your issue. Otherwise it is not possible to help

**Comment by omensight** [comment]
> @omensight please post code to reproduce your issue. Otherwise it is not possible to help

Hi, thanks, I am using Riverpod alongside EeasyLocalization:
This is a listener of a Riverpod provider that is in the Navigation view:

```
ref.listen(selectedLocalizationProvider, (previous, localization) {
      var languageCode = localization.locale.$1;
      var countryCode = localization.locale.$2;
      print('$languageCode-$countryCode');
      if (countryCode != null) {
        EasyLocalization.of(context)
            ?.setLocale(Locale(languageCode, countryCode));
      } else {
        EasyLocalization.of(context)?.setLocale(Locale(languageCode));
      }
    });
```

It effectively changes the locale to the respective language and country code like it shows in the debug console:
```
flutter: [🌎 Easy Localization] [INFO] Locale en_US changed
flutter: [🌎 Easy Localization] [INFO] Locale en_US saved
flutter: [🌎 Easy Localization] [DEBUG] Build
flutter: [🌎 Easy Localization] [DEBUG] Init Localization Delegate
flutter: [🌎 Easy Localization] [DEBUG] Init provider
```

The way I am using localizations is with the generator and translating them with:
```
LocaleKeys.installByDefault.tr()
```

And I have also tried passing the context object:

```
LocaleKeys.installByDefault.tr(context: context)
```

But nothing seems to work.

One more thing is I'm configuring EasyLocalizations in the next way:

```
runApp(ProviderScope(
    child: EasyLocalization(
        assetLoader: const CodegenLoader(),
        supportedLocales: const [
          Locale('en', 'US'),
          Locale('es'),
        ],
        fallbackLocale: const Locale('en', 'US'),
        path: 'assets/langs',
        child: const NoClickApp()),
  ));
```
Thank you so much.

**Comment by bw-flagship** [comment]
@omensight I appreciate that you posted that code. However, to rebuild it, I would  have to spend lots of time building an app around your code.
it's far better if you create a minimal sample ( a fresh app) which demonstrates your problem.

**Comment by omensight** [comment]
> @omensight I appreciate that you posted that code. However, to rebuild it, I would have to spend lots of time building an app around your code. it's far better if you create a minimal sample ( a fresh app) which demonstrates your problem.

Thanks, I have created a repository example here:
https://github.com/omensight/easy_localization_bug_example

**Comment by bw-flagship** [comment]
@omensight sorry for the late reply. It was hard for me to debug, since I have no real knowledge of riverpod and your business logic. One mistake I found was that the following line was missing:

`locale:context.locale`

But I had to remove the riverpod stuff to make it work, see this PR:
https://github.com/omensight/easy_localization_bug_example/pull/1/files

With those changes, the language updates accordingly. I hope this helps to find out how to make it work in your application!
